RoddyCore/test/de/dkfz/roddy/config/ToolFileParameterTest.groovy

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2017 eilslabs.
+ *
+ * Distributed under the MIT License (license terms are at https://www.github.com/eilslabs/Roddy/LICENSE.txt).
+ */
+
+package de.dkfz.roddy.config
+
+import de.dkfz.roddy.knowledge.files.BaseFile
+import groovy.transform.CompileStatic;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+/**
+ * Created by heinold on 08.03.17.
+ */
+@CompileStatic
+public class ToolFileParameterTest {
+    @Test
+    public void testCloneAndEquals() throws Exception {
+        def src = new ToolFileParameter(BaseFile, [], "ABC", new ToolFileParameterCheckCondition(true), "DEF", null, "abc")
+        def clone = src.clone()
+        assert clone == src
```

Comment:
Ooook... Good I hope things will not get more complicated now. I'll take a look at the post later.